fs: allow enabling filesystems with `FILE_SYSTEM_LIB_LINK`

JordanYates opened the pull request:
Update the Kconfig configuration to allow enabling the underlying filesystems (e.g. exFAT), with only `FILE_SYSTEM_LIB_LINK` enabled.

JordanYates committed:
fs: allow enabling filesystems with `FILE_SYSTEM_LIB_LINK`

Update the Kconfig configuration to allow enabling the underlying
filesystems (e.g. exFAT), with only `FILE_SYSTEM_LIB_LINK` enabled.

Signed-off-by: Jordan Yates <jordan@embeint.com>

JordanYates committed:
fs: ext2: update `CONFIG_MAX_FILES` name

Add the `EXT2` prefix to `config MAX_FILES` to more appropriately
namespace the symbol.

Signed-off-by: Jordan Yates <jordan@embeint.com>

JordanYates committed:
tests: fs: lib_link: added

Add a test validating that Zephyr can compile with the 3 underlying
filesystems enabled with only `CONFIG_FILE_SYSTEM_LIB_LINK` enabled.

Signed-off-by: Jordan Yates <jordan@embeint.com>

de-nordic commented on subsys/fs/ext2/Kconfig:
```diff
@@ -18,7 +18,7 @@ if FILE_SYSTEM_EXT2
 	visible if FILE_SYSTEM_EXT2
 
-config MAX_FILES
+config EXT2_MAX_FILES
```
This is change to Kconfig that is available to an user, there should be release note on change in Kconfig name and migration guide provided.

kartben replied:
This hasn't been addressed properly -- @de-nordic is right, and the important part is the migration guide actually. I don't think there is anything worth mentioning in release notes so please just move the bullet point you've added to release notes to migration guide instead. Thanks!

de-nordic replied:
@kartben  OK. So for API changes, etc we no longer do the release notes, rather migration guide?

JordanYates committed:
fs: ext2: fix uninitialized variable

The compiler is unaware of the `block_size > 0` precondition, so it
outputs warnings of uninitialized `reclen`. Explicitly initialize the
variable to suppress this.

Signed-off-by: Jordan Yates <jordan@embeint.com>
